lib/cylc/job_file.py

```diff
@@ -206,6 +206,10 @@ def _write_environment_1(self, handle, job_conf):
             ' '.join(job_conf['namespace_hierarchy']))
         handle.write(
             '\n    export CYLC_TASK_TRY_NUMBER=%s' % job_conf['try_num'])
+        if job_conf['param_var']:
+            for var, val in job_conf['param_var'].items():
+                value = str(val)  # (needed?)
```

Comment:
This should not be needed. The template will do the essential string conversion.